server/src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -114,20 +114,28 @@ public RecoveryTarget(
         this.snapshotFilesProvider = snapshotFilesProvider;
         this.snapshotFileDownloadsPermit = snapshotFileDownloadsPermit;
         this.shardId = indexShard.shardId();
-        final String tempFilePrefix = RECOVERY_PREFIX + UUIDs.randomBase64UUID() + ".";
-        this.multiFileWriter = new MultiFileWriter(
-            indexShard.store(),
-            indexShard.recoveryState().getIndex(),
-            tempFilePrefix,
-            logger,
-            this::ensureRefCount
-        );
+        this.multiFileWriter = createMultiFileWriter();
         this.store = indexShard.store();
         // make sure the store is not released until we are done.
         store.incRef();
         indexShard.recoveryStats().incCurrentAsTarget();
     }
 
+    private void refreshMultiFileWriter() {
```

Comment:
Can we call this "recreateMultiFileWriter" or "resetMultiFileWriter" instead? I think refresh does not match what it does.